src/ui/include/scratchpad.h

```diff
@@ -23,11 +23,45 @@
 #include <QWidget>
 
 #include <functional>
+#include <memory>
 #include <qobjectdefs.h>
 
 class QPlainTextEdit;
 class QStatusBar;
 class QLineEdit;
+class QLabel;
+class QComboBox;
+class QVariant;
+class QFormLayout;
+
+namespace klogg {
+
+class DateTimeBox : public QObject {
+  public:
+    DateTimeBox( QFormLayout* layout = nullptr );
+    ~DateTimeBox() = default;
+
+    inline QFormLayout* layout()
+    {
+        return layout_;
+    }
+
+    QString displayTime( const QString& text );
+
+  private:
+    QString displayTime();
+
+  private:
+    QFormLayout* layout_;
+    std::unique_ptr<QFormLayout> selfLayout_;
+
+    // Use the validity of QVariant to decide whether to display the time
+    std::unique_ptr<QVariant> timestamp_;
```

Comment:
Is there a reason to use `unique_ptr`? And `QVariant`? Looks like it can be just `std::optional<int64>`.